src/operator/controllers/linkerd/linkerd_manager.go

```diff
@@ -0,0 +1,763 @@
+	if err != nil {
+		ldm.recorder.RecordWarningEventf(intents, ReasonGettingLinkerdPolicyFailed, "Could not get Linkerd policies: %s", err.Error())
+		return err
+	}
+
+	err = ldm.Client.List(ctx,
+		&existingServers,
+		client.MatchingLabels{otterizev1alpha3.OtterizeLinkerdServerAnnotationKey: clientFormattedIdentity})
+	if err != nil {
+		ldm.recorder.RecordWarningEventf(intents, ReasonGettingLinkerdPolicyFailed, "Could not get Linkerd servers: %s", err.Error())
+		return err
+	}
+
+	err = ldm.Client.List(ctx,
+		&existingHttpRoutes,
+		client.MatchingLabels{otterizev1alpha3.OtterizeLinkerdServerAnnotationKey: clientFormattedIdentity})
+	if err != nil {
+		ldm.recorder.RecordWarningEventf(intents, ReasonGettingLinkerdPolicyFailed, "Could not get Linkerd http routes: %s", err.Error())
+		return err
+	}
+
+	for _, existingPolicy := range existingPolicies.Items {
+		err := ldm.Client.Delete(ctx, &existingPolicy)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, existingServer := range existingServers.Items {
+		err := ldm.Client.Delete(ctx, &existingServer)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, existingRoute := range existingHttpRoutes.Items {
+		err := ldm.Client.Delete(ctx, &existingRoute)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (ldm *LinkerdManager) deleteOutdatedResources(ctx context.Context,
+	validResources map[string]*goset.Set[types.UID],
+	existingPolicies authpolicy.AuthorizationPolicyList,
+	existingServers linkerdserver.ServerList,
+	existingRoutes authpolicy.HTTPRouteList) error {
+	for _, existingPolicy := range existingPolicies.Items {
+		if !validResources[AuthorizationPolicies].Contains(existingPolicy.UID) {
+			err := ldm.Client.Delete(ctx, &existingPolicy)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	for _, existingServer := range existingServers.Items {
+		if !validResources[Servers].Contains(existingServer.UID) {
+			err := ldm.Client.Delete(ctx, &existingServer)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	for _, existingRoute := range existingRoutes.Items {
+		if !validResources[Routes].Contains(existingRoute.UID) {
+			err := ldm.Client.Delete(ctx, &existingRoute)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (ldm *LinkerdManager) createResources(
+	ctx context.Context,
+	clientIntents *otterizev1alpha3.ClientIntents,
+	clientServiceAccount string, // supplied in the create method
+) (map[string]*goset.Set[types.UID], error) {
+	currentResources := map[string]*goset.Set[types.UID]{
+		Servers:               goset.NewSet[types.UID](),
+		AuthorizationPolicies: goset.NewSet[types.UID](),
+		Routes:                goset.NewSet[types.UID](),
+	}
+
+	for _, intent := range clientIntents.GetCallsList() {
+		if intent.Type != "" && intent.Type != otterizev1alpha3.IntentTypeHTTP && intent.Port != 0 { // this will skip non http ones, db for example, skip port doesnt exist as well
```

Comment:
The port condition seems inverted, as in if port is NOT empty AND type is not HTTP it will continue. But perhaps you meant

if intent.Type != "" // if intent type is not empty
&& intent.Type != IntentTypeHTTP // and intent type is not http
|| intent.Port == "" // OR port is empty